bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,254 @@
+            logger.trace("Response: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                if (!ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            return resp.getContentAsString();
+        } catch (InterruptedException e) {
+            logger.debug("API execution has been interrupted", e);
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (TimeoutException e) {
+            logger.debug("Timeout during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (ExecutionException e) {
+            logger.debug("Exception during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+        return null;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(VektivaSmarwiConfiguration.class);
+        logger.debug("IP address: {}", config.ip);
+
+        future = scheduler.scheduleWithFixedDelay(this::checkStatus, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private synchronized void initializeWebSocketSession() {
+        if (config.useWebSockets) {
+            closeSession();
+            session = createSession();
+            if (session != null) {
+                logger.debug("WebSocket connected!");
+            }
+        }
+    }
+
+    private void checkStatus() {
+        String url = "http://" + config.ip + "/statusn";
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            logger.debug("status values: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                processStatusResponse(resp.getContentAsString());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "got response code: " + resp.getStatus());
+            }
+            // reconnect web socket if not connected
+            if (config.useWebSockets && (session == null || !session.isOpen())
+                    && ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                logger.debug("Initializing WebSocket session");
+                initializeWebSocketSession();
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "exception during status getting");
+            logger.debug("Exception during status update", e);
+            session = null;
+        }
+    }
+
+    public synchronized void processStatusResponse(String content) {
+        if (!ThingStatus.ONLINE.equals(getThing().getStatus())) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        String[] values = content.split("\n");
```

Comment:
The usage of values doesn't seem very efficient as it's processed each time in `getProperyValue`. Maybe put it in a map first:
```
Map<String, String> values = Stream.of(content.split("\n")).map(s -> s.split(":"))
                    .filter(s -> s.length == 2).collect(Collectors.toMap(s -> s[0], s -> s[1]));
```
And than read the value 
```
private static final String NA = "N/A";
...
updateProperty("type", values.getOrDefault("t", NA));
```

Reply:
Nice solution, implemented